packages/jni-swig-stub/realm.i

```diff
@@ -96,32 +96,41 @@ return $jnicall;
 // we have a distinction (type map, etc.) in the C API that we can use for targeting the type map.
 bool realm_object_is_valid(const realm_object_t*);
 
+%{
+void throw_as_java_exception(JNIEnv *jenv) {
+    realm_error_t error;
+    if (realm_get_last_error(&error)) {
+        std::string message(error.message);
+        realm_clear_last_error();
```

Comment:
Maybe a question for the C API folks, but is there a chance of a race condition using this kind of API?, i.e. shouldn't `realm_get_last_error` also clear it?

Reply:
It's thread local, so should be ok: https://github.com/realm/realm-core/blob/master/src/realm.h#L364